applications/nrf_desktop/src/modules/dvfs.c

```diff
@@ -137,36 +147,72 @@ static void handle_dvfs_error(int32_t err)
 		timeout = DVFS_RETRY_INITIALIZATION_TIMEOUT;
 	} else {
 		LOG_ERR("DVFS freq change returned with error: %d", err);
-		module_set_state(MODULE_STATE_ERROR);
-		module_state = STATE_ERROR;
-		cancel_dvfs_retry_work();
+		dvfs_fatal_error();
 		return;
 	}
 	(void) k_work_reschedule(&dvfs_retry.retry_work, timeout);
 }
 
-static void set_dvfs_freq(enum dvfs_frequency_setting target_freq)
+static void dvfs_notify_cb(void)
 {
-	int32_t ret = dvfs_service_handler_change_freq_setting(target_freq);
+	(void) k_work_submit(&dvfs_notify_work);
+}
+
+static void dvfs_notify_work_handler(struct k_work *work)
+{
+	int res;
+	int32_t ret = sys_notify_fetch_result(&cli.notify, &res);
+
+	if (ret < 0) {
+		LOG_DBG("Work not completed");
```

Comment:
If this happens it means some other module is also using cli.notify.

There are also other events not related to dvfs: SYS_NOTIFY_METHOD_SPINWAIT, SYS_NOTIFY_METHOD_SIGNAL, SYS_NOTIFY_METHOD_CALLBACK. They should not happen in our case, but it isn't also fatal if they occurs. I would drop such situation instead of failing whole system.

Reply:
Still IMO this is not critical error and we shouldn't shutdown app when it happens but I can make it your way.

Reply:
The main question here is: Can this work be triggered when result is not ready? If not, then we should treat lack of result as a fatal error here

Reply:
For sure it is more safely to enter fatal error state. If we will encounter this error often we can think what should we done. If we do not encounter it we shouldn't bother with it much now.

Reply:
In general, triggering a fatal error for an unexpected situation may not be a bad choice (especially if the unexpected situation could break implementation of the module).